### DISCUSSION TITLE
Code that selects a column with "1 to 10" keypress goes wrong

### DISCUSSION THREAD

**Discussion by leemthai** [Textualize/textual#4854] [Q&A]
I am trying to use move_cursor() to allow users to press a "1-10" key to select relevant columns. It appears to work but then the label column can't decide whether to appear or not afterwards. Just moving the mouse around will cause it to appear/disappear erratically. It's very weird. Any idea if it is something I am doing or potentially a bug ?

![weirder](https://github.com/user-attachments/assets/a1586f2c-8417-41f8-ade0-f170fc04b328)

**Comment by TomJGooding** [comment]
I've attempted to create a quick [minimal reproducible example](https://github.com/Textualize/textual/discussions/3361) but I can't reproduce this issue.

What other updates are you doing in your code when the cursor is moved?

```python
from textual.app import App, ComposeResult
from textual.widgets import DataTable


class MyDataTable(DataTable):
    BINDINGS = [
        ("0", "move_column(0)"),
        ("1", "move_column(1)"),
        ("2", "move_column(2)"),
        ("3", "move_column(3)"),
        ("4", "move_column(4)"),
        ("5", "move_column(5)"),
        ("6", "move_column(6)"),
        ("7", "move_column(7)"),
        ("8", "move_column(8)"),
        ("9", "move_column(9)"),
    ]

    def action_move_column(self, column: int) -> None:
        self.move_cursor(column=column)

    def on_mount(self) -> None:
        self.cursor_type = "column"

        for col_num in range(10):
            self.add_column(f"C{col_num}")
        for row_num in range(10):
            self.add_row(
                *(f"C{col_num}R{row_num}" for col_num in range(10)),
                label=f"R{row_num}",
            )


class DataTableApp(App):
    def compose(self) -> ComposeResult:
        yield MyDataTable()


if __name__ == "__main__":
    app = DataTableApp()
    app.run()
```

**Comment by leemthai** [comment]
Many thanks for the demo. My code was a little overcomplicated. For example, selecting a new column was triggering a "sort on this column". I've redone it so it just works more like your demo and just selects the column and its working much better now.
